deepchem/feat/molecule_featurizers/sparse_matrix_one_hot_featurizer.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from typing import List
+
+import numpy as np
+import scipy
+from deepchem.feat.base_classes import Featurizer
+from typing import Any, Iterable
+
+logger = logging.getLogger(__name__)
+codes = [
+    'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R',
+    'S', 'T', 'V', 'W', 'Y', 'X', 'Z', 'B', 'U', 'O'
+]
+
+class SparseMatrixOneHotFeaturizer(Featurizer):
+  """Encodes any arbitrary string as a one-hot array .
+
+  Standalone Usage:
+
+  >>> import deepchem as dc
+  >>> codes = [
+      'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R',
+      'S', 'T', 'V', 'W', 'Y','X','Z','B','U','O'
+      ]
+  >>> featurizer = dc.feat.SparseMatrixOneHotFeaturizer(codes)
+  >>> sequence = "MMMQLA"
+  >>> encodings = featurizer.featurize(sequence)
+  >>> print(encodings)
+  (0, 10)	1.0
+  (1, 10)	1.0
+  (2, 10)	1.0
+  (3, 13)	1.0
+  (4, 9)	1.0
+  (5, 0)	1.0
+  >>> encodings[0].shape
+  (6, 25)
+  >>> featurizer.untransform(encodings[0])
+  'MMMQLA'
+
+  """
+
+  def __init__(
+      self,
+      charset: List[str] = codes,
+  ):
+    """Initialize featurizer.
+
+    Parameters
+    ----------
+    charset: List[str] (default code)
+      A list of strings, where each string is length 1 and unique.
+    """
+    if len(charset) != len(set(charset)):
+      raise ValueError("All values in charset must be unique.")
+    self.charset = charset
+    from sklearn.preprocessing import OneHotEncoder
+    from sklearn.preprocessing import LabelEncoder
+
+    label_encoder = LabelEncoder()
+    integer_encoded = label_encoder.fit_transform(self.charset)
+    integer_encoded = integer_encoded.reshape(len(integer_encoded), 1)
+    cat = np.array(self.charset).reshape(1, len(self.charset))
+    self.ohe = OneHotEncoder(categories=list(cat))
+
+  def featurize(self,
```

Comment:
Maybe remove and use the parent method directly

Reply:
is it possible to avoid the use of the Featurize parent method ? This generate  a problem because the datapoint is analyzed each element at the time and the sklearn OneHotFeaturizer takes just the string and not the individual character.